.github/workflows/test.yml

```diff
@@ -1,40 +1,66 @@
-      SCCACHE_GHA_ENABLED: true
-      CARGO_INCREMENTAL: 0
-    steps:
-      - name: Check out repository code
-        uses: actions/checkout@v4
-        with:
-          persist-credentials: false
-      - name: Set up sccache (part 1)
-        uses: mozilla-actions/sccache-action@v0.0.3
-        with:
-          token: ${{ secrets.GITHUB_TOKEN }}
-      - name: Set up sccache (part 2)
-        run: sccache --start-server
-      - uses: taiki-e/install-action@v2
-        with:
-          tool: just,cargo-llvm-cov,cargo-nextest
-      - name: Run tests
-        run: |
-          cd ${{ github.workspace }}
-          just check
-          RUST_BACKTRACE=1 just ci-test
-      - name: Upload coverage information
-        run: |
-          curl -Os https://uploader.codecov.io/latest/linux/codecov 
-          chmod +x codecov
-          ./codecov
+    test:
+        name: test
+        runs-on: ubuntu-latest
+        env:
+            RUSTC_WRAPPER: sccache
+            SCCACHE_GHA_ENABLED: true
+            CARGO_INCREMENTAL: 0
+        steps:
+            - name: Check out repository code
+              uses: actions/checkout@v4
+              with:
+                  persist-credentials: false
+            - uses: dtolnay/rust-toolchain@stable
+              with:
+                  components: llvm-tools, clippy, rust-src
+            - name: Set up sccache (part 1)
+              uses: mozilla-actions/sccache-action@v0.0.3
+              with:
+                  token: ${{ secrets.GITHUB_TOKEN }}
+            - name: Set up sccache (part 2)
+              run: sccache --start-server
+            - uses: taiki-e/install-action@v2
+              with:
+                  tool: just,cargo-llvm-cov,cargo-nextest
+            - name: Run tests
+              run: |
+                  cd ${{ github.workspace }}
+                  just check --all-targets
+                  RUST_BACKTRACE=1 just ci-test
+            - name: Upload coverage information
+              run: |
+                  curl -Os https://uploader.codecov.io/latest/linux/codecov
+                  chmod +x codecov
+                  ./codecov
+
+    msrv:
+        runs-on: ubuntu-latest
+        steps:
+            - uses: actions/checkout@v4
+              with:
+                  persist-credentials: false
+            - uses: dtolnay/rust-toolchain@master
+              with:
+                  toolchain: 1.75.0
+                  components: llvm-tools, clippy, rust-src
+            - name: Set up sccache (part 1)
+              uses: mozilla-actions/sccache-action@v0.0.3
+              with:
+                  token: ${{ secrets.GITHUB_TOKEN }}
+            - name: Set up sccache (part 2)
+              run: sccache --start-server
+            - uses: taiki-e/install-action@v2
+              with:
+                  tool: just
+            - name: Check library code with minimum supported Rust version
+              run: |
+                  just check
```

Comment:
I went looking for whether `cargo clippy` is equivalent to `cargo check` (it seems it is), but I think this remains wanting a `--locked` and `--lib`?